File: ganttproject/src/net/sourceforge/ganttproject/action/edit/CutAction.java
/*
GanttProject is an opensource project management tool.
Copyright (C) 2005-2011 GanttProject Team

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 3
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package net.sourceforge.ganttproject.action.edit;

import net.sourceforge.ganttproject.action.GPAction;
import net.sourceforge.ganttproject.gui.UIUtil;
import net.sourceforge.ganttproject.gui.view.GPViewManager;
import net.sourceforge.ganttproject.undo.GPUndoManager;

import java.awt.event.ActionEvent;
import java.beans.PropertyChangeEvent;
import java.beans.PropertyChangeListener;

//TODO Enable/Disable action on selection changes
public class CutAction extends GPAction {
  private final GPViewManager myViewmanager;
  private final GPUndoManager myUndoManager;

  public CutAction(GPViewManager viewManager, GPUndoManager undoManager) {
    super("cut");
    myViewmanager = viewManager;
    myUndoManager = undoManager;
  }

  @Override
  public void actionPerformed(ActionEvent e) {
    if (calledFromAppleScreenMenu(e)) {
      return;
    }
    myUndoManager.undoableEdit(getLocalizedName(), new Runnable() {
      @Override
      public void run() {
        myViewmanager.getSelectedArtefacts().startMoveClipboardTransaction();
      }
    });
  }

  @Override
  public CutAction asToolbarAction() {
    final CutAction result = new CutAction(myViewmanager, myUndoManager);
    result.setFontAwesomeLabel(UIUtil.getFontawesomeLabel(result));
    this.addPropertyChangeListener(new PropertyChangeListener() {
      @Override
      public void propertyChange(PropertyChangeEvent evt) {
        if ("enabled".equals(evt.getPropertyName())) {
          result.setEnabled((Boolean)evt.getNewValue());
        }
      }
    });
    result.setEnabled(this.isEnabled());
    return result;
  }
}

File: ganttproject/src/main/java/net/sourceforge/ganttproject/action/edit/CutAction.kt
/*
GanttProject is an opensource project management tool.
Copyright (C) 2005-2011 GanttProject Team

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 3
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package net.sourceforge.ganttproject.action.edit

import net.sourceforge.ganttproject.action.GPAction
import net.sourceforge.ganttproject.gui.UIFacade
import net.sourceforge.ganttproject.gui.UIUtil
import net.sourceforge.ganttproject.gui.view.GPViewManager
import net.sourceforge.ganttproject.undo.GPUndoManager
import java.awt.event.ActionEvent

//TODO Enable/Disable action on selection changes
class CutAction(
    private val myViewmanager: GPViewManager,
    private val myUndoManager: GPUndoManager,
    private val myUiFacade: UIFacade
) : GPAction("cut") {
    override fun actionPerformed(e: ActionEvent) {
        if (calledFromAppleScreenMenu(e)) {
            return
        }
        //    myUndoManager.undoableEdit(getLocalizedName(), new Runnable() {
//      @Override
//      public void run() {
        myViewmanager.selectedArtefacts.startMoveClipboardTransaction()
        //      }
        //});
        myUiFacade.activeChart.focus()
    }

    override fun asToolbarAction(): CutAction {
        val result = CutAction(myViewmanager, myUndoManager, myUiFacade)
        result.setFontAwesomeLabel(UIUtil.getFontawesomeLabel(result))
        this.addPropertyChangeListener { evt ->
            if ("enabled" == evt.propertyName) {
                result.isEnabled = (evt.newValue as Boolean)
            }
        }
        result.isEnabled = this.isEnabled
        return result
    }
}
